pkg/cmd/gist/create/create.go

```diff
@@ -153,6 +153,12 @@ func createRun(opts *CreateOptions) error {
 			if httpError.OAuthScopes != "" && !strings.Contains(httpError.OAuthScopes, "gist") {
 				return fmt.Errorf("This command requires the 'gist' OAuth scope.\nPlease re-authenticate by doing `gh config set -h github.com oauth_token ''` and running the command again.")
 			}
+			if httpError.StatusCode == http.StatusUnprocessableEntity {
+				if detectEmptyFiles(files) {
+					fmt.Fprintf(errOut, "%s Failed to create gist: %s", cs.Red("X"), errors.New("A gist file must not contain only whitespace characters"))
+					return nil
```

Comment:
If you are printing an error manually to stderr, you should still return an error object: `cmdutil.SilentError`